tests/core/actions/test_forms.py

```diff
@@ -1879,3 +1881,61 @@ async def test_extract_slots_with_mapping_conditions_during_form_activation():
     )
 
     assert form_events == expected_events
+
+
+async def test_check_if_slot_validation_happens_twice(caplog: LogCaptureFixture):
```

Comment:
Please add a docstring to link this unit test to the Github ticket as context for future developers.
```suggestion
async def test_form_validation_happens_once(caplog: LogCaptureFixture):
```